code/game/gamemodes/miniantags/abduction/abduction_gear.dm

```diff
@@ -337,18 +337,24 @@ CONTENTS:
 
 	add_attack_logs(user, L, "Stunned with [src]")
 
-/obj/item/abductor_baton/proc/SleepAttack(mob/living/L,mob/living/user)
-	if(L.IsStunned() || L.IsSleeping())
-		L.visible_message("<span class='danger'>[user] has induced sleep in [L] with [src]!</span>", \
+/obj/item/abductor_baton/proc/SleepAttack(mob/living/L, mob/living/user)
+	var/mob/living/carbon/C = L
+	if(!iscarbon(L))
+		return
+	if(C.getStaminaLoss() < 100)
+		C.AdjustDrowsy(2 SECONDS)
+		to_chat(user, "<span class='warning'>Sleep inducement works fully only on stunned or asleep specimens!</span>")
+		C.visible_message("<span class='danger'>[user] tried to induce sleep in [L] with [src]!</span>", \
+						"<span class='userdanger'>You suddenly feel drowsy!</span>")
+		return
+	if(do_mob(user, C, 2.5 SECONDS))
+		C.visible_message("<span class='danger'>[user] has induced sleep in [L] with [src]!</span>", \
 							"<span class='userdanger'>You suddenly feel very drowsy!</span>")
 		playsound(loc, 'sound/weapons/egloves.ogg', 50, TRUE, -1)
-		L.Sleeping(120 SECONDS)
-		add_attack_logs(user, L, "Put to sleep with [src]")
+		C.Sleeping(120 SECONDS)
+		add_attack_logs(user, C, "Put to sleep with [src]")
 	else
-		L.AdjustDrowsy(2 SECONDS)
-		to_chat(user, "<span class='warning'>Sleep inducement works fully only on stunned specimens!</span>")
-		L.visible_message("<span class='danger'>[user] tried to induce sleep in [L] with [src]!</span>", \
-							"<span class='userdanger'>You suddenly feel drowsy!</span>")
+		return
```

Comment:
this `else return` can be removed entirely

Reply:
removed, works upon testing.